provider/resource_service.go

```diff
@@ -144,262 +129,565 @@ func getKpiBSData(ctx context.Context, cc models.ClientConfig, id string) (map[s
+								},
+							},
+						},
+					},
 				},
 			},
-			"service_depends_on": {
-				Type:        schema.TypeSet,
-				Optional:    true,
+			"service_depends_on": schema.SetNestedBlock{
 				Description: "A set of service descriptions with KPIs in those services that this service depends on.",
-				Elem: &schema.Resource{
-					Schema: map[string]*schema.Schema{
-						"service": {
-							Type:        schema.TypeString,
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						"service": schema.StringAttribute{
 							Required:    true,
 							Description: "_key value of service that this service depends on.",
 						},
-						"kpis": {
-							Type:        schema.TypeSet,
+						"kpis": schema.SetAttribute{
 							Required:    true,
 							Description: "A set of _key ids for each KPI in service identified by serviceid, which this service will depend on.",
-							Elem: &schema.Schema{
-								Type: schema.TypeString,
-							},
+							ElementType: types.StringType,
 						},
-						"overloaded_urgencies": {
-							Type:        schema.TypeMap,
+						"overloaded_urgencies": schema.MapAttribute{
 							Optional:    true,
 							Description: "A map of urgency overriddes for the KPIs this service is depending on.",
-							Elem: &schema.Schema{
-								Type: schema.TypeInt,
-							},
+							ElementType: types.Int64Type,
 						},
 					},
 				},
 			},
-			"tags": {
-				Type:        schema.TypeSet,
-				Optional:    true,
-				Description: "The tags for the service. Cannot contain the following characters: / \\ \" ' ! @ ? . , ; $ ^ ",
-				Elem: &schema.Schema{
-					Type: schema.TypeString,
+		},
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
 				},
 			},
-			"kpi": {
-				Type:        schema.TypeSet,
+			"title": schema.StringAttribute{
+				Required:    true,
+				Description: "Title of the service.",
+			},
+			"description": schema.StringAttribute{
 				Optional:    true,
-				Description: "A set of KPI descriptions for this service.",
-				Elem: &schema.Resource{
-					Schema: kpisSchema,
-				},
+				Description: "User-defined description for the service.",
+			},
+			"enabled": schema.BoolAttribute{
+				Optional:    true,
+				Description: "Boolean value defining whether the service should be enabled.",
+			},
+			"is_healthscore_calculate_by_entity_enabled": schema.BoolAttribute{
+				Optional:    true,
+				Description: "Set the Service Health Score calculation to account for the severity levels of individual entities if at least one KPI is split by entity.",
+			},
+			"security_group": schema.StringAttribute{
+				Optional:    true,
+				Description: "The team the object belongs to.",
+			},
+			"tags": schema.SetAttribute{
+				Optional:    true,
+				Description: "The tags for the service.",
+				ElementType: types.StringType,
```

Comment:
https://github.com/TiVo/terraform-provider-splunk-itsi/pull/418/files#diff-6bcac826cdcc1c2d664f422c46abbafc7b31fd82f90dfdb8ca063134fa250af2R531

Reply:
I'll test with defaults as well, I didn't realize it could be set for collections, just remembered it was not possible for sdkv2

Reply:
Yes, it's possible in the new framework, for example: https://github.com/TiVo/terraform-provider-splunk-itsi/blob/v2-dev/provider/resource_entity.go#L242

Can do similar for maps, too.

Reply:
But I was actually talking more generally about all the optional fields, not just tags. From what I remember it's usually better to set the default values even for simple types like Strings, etc. That should help eliminate drift when terraform config changes between `""` <-> `null`, for example..

Reply:
But I also know that the new terraform framework has some bugs related to computed + optional fields, unfortunately. And sometimes we can't really use them without triggering those 🐛 s...

Reply:
Sorry, I re-reading in the morning, and realized, I inserted a bad link. I tried to point to:
https://github.com/TiVo/terraform-provider-splunk-itsi/blob/73c062c4f3836bbef3917ce01bcc52516c0ab051/provider/resource_service.go#L531
Thet in case len=0 I set NullSet, thank you, and I'll test with default :)

Reply:
Added defaults to everything, except empty string. It stands for the import issue also. If an empty string is specified it is mapped to null in configuration (even in case the passed state contains an empty string inside), and then it tries to compare null with "", and doesn't print that (nothing like description: null -> ""), just results in very unobvious plan